rust/origen/src/core/pattern/pinaction.rs

```diff
@@ -0,0 +1,41 @@
+//! Defines the set of actions associated with a pattern pin action
+pub use super::operation::Operation;
+
+pub struct PinAction {
+    name: String,
```

Comment:
I think in the end this should be an immutable reference to a pin/group. I would guess the reference will give a smaller memory footprint and allow more flexibility when processing the actions.